src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SqlServerType.cs

```diff
@@ -41,7 +42,17 @@ public class ServerVersionHelper
         /// </summary>
         public static ValidForFlag GetValidForFlag(SqlServerType serverType, Database database = null)
         {
-            return GetValidForFlag(serverType, database != null && database.IsSqlDw);
+            var isSqlDw = false;
+            try
+            {
+                isSqlDw = database.IsSqlDw;
+            }
+            catch (Exception e)
```

Comment:
what exceptions are we expecting here? if recoverable, we should log it as information and add explanation says this is expected so that it won't cause false alarms.